client/src/components/selects/addResourceSelect.tsx

```diff
@@ -5,96 +5,125 @@ import { IconName } from '@blueprintjs/icons';
 
 import TSelect from './TSelect';
 
-interface ISelectProps {
+interface Props {
   disabled?: boolean;
   icon?: IconName;
-  inputItem: string;
+  inputItem?: Resource;
   intent?: Intent;
-  items?: string[];
+  items?: Resource[];
   loading?: boolean;
   onChange: any;
   popoverProps?: any;
 }
 
-export default class AddResourceSelect extends React.Component<
-  ISelectProps,
-  any
-> {
-  private renderItem: ItemRenderer<string> = (
-    resource: string,
+interface Profile {
+  id: string;
+  name: string;
+  type: string;
+}
+interface Resource {
+  id: string;
+  name: string;
+  type: string;
+  profiles: Profile[];
+}
+
+const AddResourceSelect = ({
+  disabled,
+  icon,
+  inputItem,
+  intent,
+  items,
+  loading,
+  onChange
+}: Props) => {
+  const renderMenuItem: ItemRenderer<Profile> = (
+    profile: Profile,
     { handleClick, modifiers, query }
   ) => {
     return (
       <MenuItem
-        key={resource}
+        key={profile.id}
         onClick={handleClick}
         text={
           <span>
-            <strong>{resource}</strong>
+            <strong>{profile.name}</strong>
           </span>
         }
       />
     );
   };
 
-  private filterByName: ItemPredicate<string> = (query, resource: string) => {
-    return resource.toLowerCase().indexOf(query.toLowerCase()) >= 0;
+  const renderItem: ItemRenderer<Resource> = (
```

Comment:
It's a detail but I think it could be nice to have something to distinguish the base resource from the others, or at least to be sure that it's at the top of the list

Reply:
good point, I'll add that

Reply:
done